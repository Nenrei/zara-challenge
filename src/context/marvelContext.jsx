import React, { createContext, useState } from 'react';

const MarvelContext = createContext({});

const MarvelContextProvider = ({ defaultValue, children }) => {
    const [favoriteCharacters, setFavoriteCharacters] = useState([]);

    const isFavoriteCharacter = (id) => {
        return favoriteCharacters.find((el) => el.id === id) != null;
    };

    let value = {
        states: {
            favorite: {
                favoriteCharacters,
                setFavoriteCharacters,
            },
        },
        actions: {
            isFavoriteCharacter,
        },
    };

    if (defaultValue) {
        value = { ...value, ...defaultValue };
    }

    return <MarvelContext.Provider value={value}>{children}</MarvelContext.Provider>;
};

const useMarvelContext = () => {
    const context = React.useContext(MarvelContext);
    if (context === undefined) {
        throw new Error('useMarvelContext must be used within a MarvelContextProvider');
    }
    return context;
};

export { MarvelContextProvider, useMarvelContext };
