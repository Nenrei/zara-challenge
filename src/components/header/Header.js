import React from 'react';
import styles from './header.modules.css';
import { Link } from 'react-router-dom';
import { useMarvelContext } from '../../context/marvelContext';
import logo from '../../../public/assets/images/logo.png';
import Icon, { ICON_TYPES } from '../icon/Icon';

const Header = () => {
    const {
        states: {
            favorite: { favoriteCharacters },
        },
    } = useMarvelContext();

    return (
        <header className={styles.header}>
            <Link to={'/'} className={styles.header__logo}>
                <img src={logo} alt="logo" />
            </Link>
            <Link to={'/favorites'} className={styles['header__fav-count']}>
                <div className={styles['header__fav-count__icon']}>
                    <Icon type={ICON_TYPES.HEART} />
                </div>
                <div className={styles['header__fav-count__text']}>{favoriteCharacters.length}</div>
            </Link>
        </header>
    );
};

export default Header;
