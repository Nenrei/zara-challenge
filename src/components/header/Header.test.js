import renderer from 'react-test-renderer';
import Header from './Header';
import { MarvelContextProvider } from '../../context/marvelContext';
import { BrowserRouter } from 'react-router-dom';

it('test de header', () => {
    const component = renderer.create(
        <MarvelContextProvider
            defaultValue={{
                states: {
                    favorite: {
                        favoriteCharacters: [],
                    },
                },
            }}
        >
            <BrowserRouter>
                <Header></Header>
            </BrowserRouter>
        </MarvelContextProvider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
