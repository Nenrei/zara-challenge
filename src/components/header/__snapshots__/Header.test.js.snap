// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test de header 1`] = `
<header
  className="header"
>
  <div
    className="header__logo"
    onClick={[Function]}
  >
    <img
      alt="logo"
      src="test-file-stub"
    />
  </div>
  <div
    className="header__fav-count"
    onClick={[Function]}
  >
    <div
      className="icon icon--heart header__fav-count__icon"
    />
    <div
      className="header__fav-count__text"
    >
      0
    </div>
  </div>
</header>
`;
